client/src/components/layout/productsPage/productsList.tsx

```diff
@@ -0,0 +1,48 @@
+import { useCallback, useMemo, useState } from "react";
+import VirtualScroll from "react-dynamic-virtual-scroll";
+import { productsListState } from "../../../store/products";
+import { useRecoilValue } from "recoil";
+import { Product } from "../../../models/product";
+import "./productsList.scss"
+import { ProductListItem } from "./productListItem";
+
+type Props = {
+    searchString: string;
+};
+
+export function ProductsList(props: Props) {
+
+    const products: Product[] = useRecoilValue(productsListState);
+    const [selectedId, setSelectedId] = useState(0);
+
+    const handleProductSelection = useCallback((id: number) => {
+        selectedId === id ? setSelectedId(0) : setSelectedId(id)
+    }, [selectedId])
+
+    const filteredProducts = useMemo(() => {
+        return products.filter(product => product.name.toLowerCase().includes(props.searchString.toLowerCase()));
+    }, [props.searchString, products]);
+
+    const renderItem = useCallback((rowIndex: number) => {
+        if (filteredProducts && filteredProducts[rowIndex]) {
+            let className = selectedId === filteredProducts[rowIndex].id ? "selected" : "list-item"
```

Comment:
```suggestion
            const className = selectedId === filteredProducts[rowIndex].id ? "selected" : "list-item"
```
we use `const` for variables that have a constant value (value that we know is not going or suppose to change)

Reply:
true, fixed!